Permissions on Trento configuration files + review and update installation with systemd steps (#251)
trento-project/docs: trento/adoc/trento-systemd-install.adoc

CREATE DATABASE trento;
CREATE DATABASE trento_event_store;
----

. Create the users:
+
[NOTE]
====

> The following commands use `wanda_user`, `wanda_password`, `trento_user`, and `trento_password` as example values. For a secure environment, replace the passwords (and user names, if needed) with your own secure values.

Reviewer: I'd be more prescriptive. Or, at least, remove the repetition "For a _secure_ (...) _secure_ values."

```suggestion
The values `wanda_user`, `wanda_password`, `trento_user`, and `trento_password` are placeholders. Replace them with your own secure values before running these commands.
```

Author: Great point, thank you  :+1:

> ====

Author: Added this info so that it is clear which values in the command can be changed

Reviewer 2: Great idea!

+
[source,bash]
----
sudo zypper install rabbitmq-server
----

. Allow connections from external hosts by modifying

> `+/etc/rabbitmq/rabbitmq.conf+`, so the {trento}-agent can reach RabbitMQ:

Reviewer: Is this path also ok in SLES 16?

Author: It worked for me on SLES 16 :sweat_smile: 
edit: sles4sap 16, I have not actually tested sles, my bad

. Create a new RabbitMQ user:
+
[source,bash,subs="attributes"]
----

> {sudo} rabbitmqctl add_user TRENTO_USER TRENTO_USER_PASSWORD

Reviewer 2: Maybe we should add a note here, similar to the one that you have added for postgresql, informing the user that trento_user and trento_user_password are example values?

----
====

. Create the `/etc/trento/trento-web` configuration file. For example:
+
[source,bash]
----
# /etc/trento/trento-web
AMQP_URL=amqp://TRENTO_USER:TRENTO_USER_PASSWORD@localhost:5672/vhost
DATABASE_URL=ecto://TRENTO_USER:WEB_PASSWORD@localhost/trento

> EVENTSTORE_URL=ecto://TRENTO_USER:WEB_PASSWORD@localhost/trento_event_store

Reviewer 2: Similar comment as the one for rabbitmq. Should we add a now explaining that the values of certain parameters are just example values?

----
====

> . Create the `/etc/trento/trento-web` configuration file. For example:

Reviewer: This also depends on SLES 15 vs 16, no?  (like `usr/etc/trento/trento-web`)

Author: Our docs right now did not mention the /usr/ location for server installation, only for agent, for 16 it worked for me with `/etc/trento/` folder, I will test it with 15 too

Author: I haven't found anything in the code about `/usr/etc/trento/` location in web, found only this: https://github.com/trento-project/web/blob/04a64420235cb0e96cfc76a00c05b8517fe468be/packaging/suse/rpm/trento-web.spec#L78-L81 that points to /etc/trento/ ; same with `etc/rabbitmq/`

Reviewer 2: @vicenteqa has also tested the manual installation processes on both SLES 15 and 16 and he has not raised a flag. Therefore let's leave it as it is for now. If the paths are adjusted in future releases, we can always align the documentation afterwards.

+
[source,bash]
----

> # /etc/trento/trento-web

Reviewer: ditto https://github.com/trento-project/docs/pull/251/changes#r3659022388

in the web interface.
====

** The `ADMIN_PASSWORD` variable must meet the following requirements:
	
*** Minimum of 8 characters
*** The password  not contain 3 consecutive identical numbers or letters (for example, 111 or aaa)
*** The password must not contain 4 consecutive  numbers or letters (for example, 1234, abcd, ABCD)

** The `ENABLE_ALERTING` enables the 

> https://www.trento-project.io/docs/web/Alerting/alerting.html[alerting system to receive email notifications]. Set `ENABLE_ALERTING` to `true` and add additional variables to the `/etc/trento/trento-web` to enable the feature:

Reviewer: ditto https://github.com/trento-project/docs/pull/251/changes#r3659022388

SMTP_USER=<<SMTP_USER>>
SMTP_PASSWORD=<<SMTP_PASSWORD>>
----

. Create the `/etc/trento/trento-wanda` configuration file. For example:
+
[source,bash]
----
# /etc/trento/trento-wanda
CORS_ORIGIN=http://localhost
AMQP_URL=amqp://TRENTO_USER:TRENTO_USER_PASSWORD@localhost:5672/vhost

> DATABASE_URL=ecto://WANDA_USER:WANDA_PASSWORD@localhost/wanda

Reviewer 2: Same suggestion: https://github.com/trento-project/docs/pull/251/changes#r3664365174.

OAS_SERVER_URL=https://trento.example.com/wanda
AUTH_SERVER_URL=http://localhost:4000
----

. Secure the configuration files.
+
To protect sensitive information, the configuration files must have `600` permissions so that only the `root` user can read them:
+
[source,bash]
----
sudo chmod 600 /etc/trento/trento-web
sudo chmod 600 /etc/trento/trento-wanda

> ----

Author: New addition

===== Prepare SSL certificate for NGINX

Create or provide a certificate for https://nginx.org/en/[NGINX] to
enable SSL for {trento}. To create the certificate, you can choose one of the following methods:

* To manually generate a self-signed certificate, see <<option-1-creating-a-self-signed-certificate>>.

> * To automate obtaining a signed certificate with Let's Encrypt, see <<option-2-using-lets-encrypt-for-a-signed-certificate-using-packagehub-repository>>.

Author: Added this info to differentiate between the two options we document

Reviewer 2: Awesome! In theory there is a third option, which is the customer providing a signed certificate entirely on their own :-)

----
openssl req -newkey rsa:2048 --nodes -keyout trento.key -x509 -days 5 -out trento.crt -addext "subjectAltName = DNS:trento.example.com"
----

. Copy the generated `+trento.key+` to a location accessible by NGINX:
+
[source,bash]
----

> sudo cp trento.key /etc/ssl/private/trento.key

Reviewer: same question about SLES 16 (applies to other places as well)
